extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/services.repositories.js

```diff
@@ -161,8 +161,48 @@ angular.module('PaperUI.services.repositories', []).factory('bindingRepository',
     var repository = new Repository($q, $rootScope, itemService, 'items')
     $rootScope.data.items = [];
     return repository;
-}).factory('ruleRepository', function($q, $rootScope, ruleService) {
-    var repository = new Repository($q, $rootScope, ruleService, 'rules')
+}).factory('ruleRepository', function($q, $rootScope, ruleService, eventService) {
+    var repository = new Repository($q, $rootScope, ruleService, 'rules', true)
     $rootScope.data.rules = [];
+
+    eventService.onEvent('smarthome/rules/*/updated', function(topic, ruleUpdate) {
+
+        var existing = repository.find(function(rule) {
+            return rule.uid === ruleUpdate[0].uid;
+        });
+        if (existing) {
+            existing.name = ruleUpdate[0].name;
+            existing.description = ruleUpdate[0].description;
+            existing.triggers = ruleUpdate[0].triggers;
+            existing.actions = ruleUpdate[0].actions;
+            existing.conditions = ruleUpdate[0].conditions;
+        }
+    });
+
+    eventService.onEvent('smarthome/rules/*/added', function(topic, rule) {
+        repository.add(rule);
+    });
+
+    eventService.onEvent('smarthome/rules/*/removed', function(topic, removedRule) {
+        var existing = repository.find(function(rule) {
+            return rule.uid === removedRule.uid;
+        });
+        repository.remove(existing);
+    });
+
+    eventService.onEvent('smarthome/rules/*/state', function(topic, rule) {
+        var existing = repository.find(function(rule) {
+            return rule.uid === topic.split('/')[2];
+        });
+        existing.status = existing.status ? existing.status : {};
+        existing.status.status = rule.status;
+        existing.status.statusDetails = rule.status;
```

Comment:
Why set we status.status to rule.status and stats.statusDetails also to rule.status? Shouldn't the details point to the details?